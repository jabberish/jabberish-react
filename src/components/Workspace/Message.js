import React from 'react';
import PropTypes from 'prop-types';
import { ListItem, Typography } from '@material-ui/core';

const Message = ({ messageData }) => {
  return (
    <ListItem>
      <Typography type="body1">
        {messageData.user.username + ': ' + messageData.text}
      </Typography>
    </ListItem>
  );
};

Message.propTypes = {
  messageData: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    user: PropTypes.object.isRequired
  }).isRequired
};

export default Message;
