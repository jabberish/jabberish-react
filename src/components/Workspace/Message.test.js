/* eslint-disable no-console */
import React from 'react';
import { shallow } from 'enzyme';
import Message from './Message';

describe('Message component', () => {
  it('renders Message', () => {
    const wrapper = shallow(<Message 
      i={1} 
      messageData={{ _id: 'some-id', text: 'text', user: {} }} 
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
