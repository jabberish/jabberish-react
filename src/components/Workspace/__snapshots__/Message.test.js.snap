// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component renders Message 1`] = `
<WithStyles(ForwardRef(ListItem))
  style={null}
>
  <WithStyles(ForwardRef(Typography))
    type="body1"
  >
    undefined: text
  </WithStyles(ForwardRef(Typography))>
</WithStyles(ForwardRef(ListItem))>
`;
